kolibri/core/logger/api.py

```diff
@@ -1053,3 +1057,68 @@ class AttemptLogViewSet(ReadOnlyValuesViewset):
     filter_class = AttemptFilter
 
     values = attemptlog_values
+
+
+class GenerateCSVLogRequestSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = GenerateCSVLogRequest
+        fields = [
+            "facility",
+            "log_type",
+            "selected_start_date",
+            "selected_end_date",
+            "date_requested",
+        ]
+
+
+class GenerateCSVLogRequestFilter(FilterSet):
+    log_type = ChoiceFilter(choices=GenerateCSVLogRequest.LOG_TYPE_CHOICES)
+    facility = ModelChoiceFilter(queryset=Facility.objects.all())
+
+    class Meta:
+        model = GenerateCSVLogRequest
+        fields = ["log_type", "facility"]
+
+
+class GenerateCSVLogRequestViewSet(viewsets.ModelViewSet):
+    """
+    API endpoint that allows csv log request data to be created or updated
+    """
+
+    permission_classes = (KolibriAuthPermissions,)
+    queryset = GenerateCSVLogRequest.objects.all()
+    serializer_class = GenerateCSVLogRequestSerializer
+    filter_class = GenerateCSVLogRequestFilter
+
+    def _get_or_create_logrequest(
+        self,
+        facility,
+        log_type,
+        selected_start_date,
+        selected_end_date,
+        date_requested,
+    ):
+
+        try:
+            csvlogrequest = GenerateCSVLogRequest.objects.get(
+                facility=facility,
+                log_type=log_type,
+            )
+            updated_fields = (
+                "selected_start_date",
+                "selected_end_date",
+                "date_requested",
+            )
+            csvlogrequest.selected_start_date = selected_start_date
+            csvlogrequest.selected_end_date = selected_end_date
+            csvlogrequest.date_requested = date_requested
+            csvlogrequest.save(update_fields=updated_fields)
+        except GenerateCSVLogRequest.DoesNotExist:
+            csvlogrequest = GenerateCSVLogRequest.objects.create(
+                facility=facility,
+                log_type=log_type,
+                selected_start_date=selected_start_date,
+                selected_end_date=selected_end_date,
+                date_requested=date_requested,
+            )
+            csvlogrequest.save()
```

Comment:
So it seems there's only ever at most 1 record for each facility and log type combination?

Reply:
Yes, with this record, the previously selected date range for that generated session and or summary log file can be retrieved and used for download.